package/src/components/Message/MessageSimple/MessageBounce.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useCallback } from 'react';
+import { Alert } from 'react-native';
+
+import {
+  MessageContextValue,
+  useMessageContext,
+} from '../../../contexts/messageContext/MessageContext';
+import {
+  MessagesContextValue,
+  useMessagesContext,
+} from '../../../contexts/messagesContext/MessagesContext';
+
+import type { DefaultStreamChatGenerics } from '../../../types/types';
+
+export type MessageBouncePropsWithContext<
+  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
+> = Pick<
+  MessagesContextValue<StreamChatGenerics>,
+  'setEditingState' | 'removeMessage' | 'retrySendMessage'
+> &
+  Pick<MessageContextValue<StreamChatGenerics>, 'message'> & {
+    isBounceDialogOpen?: boolean;
+    setIsBounceDialogOpen?: React.Dispatch<React.SetStateAction<boolean>>;
+  };
+
+export const MessageBounceWithContext = <
+  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
+>(
+  props: MessageBouncePropsWithContext<StreamChatGenerics>,
+) => {
+  const { message, removeMessage, retrySendMessage, setEditingState, setIsBounceDialogOpen } =
+    props;
+
+  const handleEditMessage = useCallback(() => {
+    setEditingState(message);
+    if (setIsBounceDialogOpen) {
+      setIsBounceDialogOpen(false);
+    }
+  }, [message, setEditingState]);
+
+  const handleResend = useCallback(() => {
+    retrySendMessage(message);
+    if (setIsBounceDialogOpen) {
+      setIsBounceDialogOpen(false);
+    }
+  }, [message, retrySendMessage]);
+
+  const handleRemoveMessage = useCallback(() => {
+    removeMessage(message);
+    if (setIsBounceDialogOpen) {
+      setIsBounceDialogOpen(false);
+    }
+  }, [message, removeMessage]);
```

Comment:
all the `useCallback` is unnecessary here.. as Alert is not a memoised component.. so you can remove them